src/useFetch.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+const useFetch = (url) => {
+    // when we succesfully fetch the data, we gonna update the state with the data
+    const [data, setData] = useState(null);
+    const [isPending, setIsPending] = useState(true);
+    const [error, setError] = useState(null)
+
+    useEffect(() => {
+        // Create abort the process functionality
+        const abortCont = new AbortController();
+
+        // using fetch request here run on intial render
+        // associte the abort controller with the fetch so we can call it and abort the fetch
+        fetch(url, { signal: abortCont.signal })
+        // when the respones resolve, we get res object this is not the data just a respone object, in order to get the data we have to do something with this object
+            .then(res => {
+                if(!res.ok){
+                    throw Error('could not fetch')
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/7bel6/b-blog/pullRequest?prid=7919486)
```suggestion
                    throw Error("could not fetch")
```